src/ipred_prepare.rs

```diff
@@ -182,177 +193,131 @@ pub unsafe fn rav1d_prepare_intra_edges<BD: BitDepth>(
             || av1_intra_prediction_edges[mode as usize]
                 .needs
                 .contains(Needs::LEFT)
-                && have_left == 0)
+                && !have_left)
     {
-        if !prefilter_toplevel_sb_edge.is_null() {
-            dst_top = &*prefilter_toplevel_sb_edge.offset((x * 4) as isize) as *const BD::Pixel;
+        let px_have = cmp::min(8 * tw, 4 * (w - x)) as usize;
+        let n = px_have + have_left as usize;
+        if let Some(prefilter_toplevel_sb_edge) = prefilter_toplevel_sb_edge {
+            let offset = (x * 4) as usize - have_left as usize;
```

Comment:
Is the `- have_left as usize` done here to avoid a `.offset(-1)` later on?

Reply:
Yes